contracts/MultiFeeDistributionFactory.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.12;
+
+import { MultiFeeDistribution } from "./MultiFeeDistribution.sol";
+import { IMultiFeeDistributionFactory } from "interfaces/IMultiFeeDistributionFactory.sol";
+import { IOwnable } from "interfaces/IOwnable.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+
+contract MultiFeeDistributionFactory is IMultiFeeDistributionFactory, Ownable {
+    bytes32 public override constant bytecodeHash =
+        keccak256(type(MultiFeeDistribution).creationCode);
+
+    // This called in the MultiFeeDistribution constructor
+    bytes public override cachedDeployData;
+
+    mapping(address => address) public override vaultToStaker;
+
+    constructor() {}
+
+    function deployStaker(address ichiVault) external override returns (address staker) {
+        bytes memory _deployData = abi.encode(ichiVault);
```

Comment:
I suspect this was more useful when the deployed contract was upgradeable and could not set values in the `constructor`. The initializer probably called back to pick up the value. 

It appears that it's only read and written by the deployed contract, so it could be passed in as a `constructor` arg. Simpler and more gas efficient.

Reply:
Including it in the constructor messes up the deterministic create2 calculation(at least from what I recall a long time ago), it would require a different computation than the standard salt, init code hash and factory address.